### PR TITLE
env api

### PR DESCRIPTION
**Pull request by doktoric**

### CODE REVIEW

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java [hunk 1]
```diff
@@ -28,66 +27,66 @@
 import io.swagger.annotations.Api;
 import io.swagger.annotations.ApiOperation;
 
-@Path("/v3/{workspaceId}/environments")
+@Path("/v4/{workspaceId}/environments")
 @Consumes(MediaType.APPLICATION_JSON)
-@Api(value = "/v3/{workspaceId}/environments", description = ControllerDescription.ENVIRONMENT_V3_DESCRIPTION, protocols = "http,https")
-public interface EnvironmentV3Endpoint {
+@Api(value = "/v4/{workspaceId}/environments", description = ControllerDescription.ENVIRONMENT_V3_DESCRIPTION, protocols = "http,https")
```
Comment (topolyai5) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: rename ENVIRONMENT_V3_DESCRIPTION to ENVIRONMENT_V4_DESCRIPTION

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java [hunk 2]
```diff
@@ -28,66 +27,66 @@
...
+public interface EnvironmentV4Endpoint {
 
     @POST
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.CREATE, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "createEnvironment")
-    DetailedEnvironmentResponse create(@PathParam("workspaceId") Long workspaceId, @Valid EnvironmentRequest request);
+    DetailedEnvironmentV4Response create(@PathParam("workspaceId") Long workspaceId, @Valid EnvironmentRequest request);
```
Comment (topolyai5) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: rename to post

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java [hunk 3]
```diff
@@ -28,66 +27,66 @@
...
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.DETACH_RESOURCES, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "detachResourcesFromEnvironment")
-    DetailedEnvironmentResponse detachResources(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
-            @Valid EnvironmentDetachRequest request);
+    DetailedEnvironmentV4Response detach(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
+                                                  @Valid EnvironmentDetachRequest request);
 
     @PUT
     @Path("/{name}/core/src/test/java/com/sequenceiq/cloudbreak/service/environment/EnvironmentServiceTest.java")
```
Comment (topolyai5) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: nice path :D

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java [hunk 4]
```diff
@@ -28,66 +27,66 @@
...
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.CHANGE_CREDENTIAL, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "changeCredentialInEnvironment")
-    DetailedEnvironmentResponse changeCredential(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
-            EnvironmentChangeCredentialRequest request);
+    DetailedEnvironmentV4Response changeCredential(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
+        @Valid EnvironmentChangeCredentialRequest request);
 
     @PUT
     @Path("/{name}/registerDatalake")
```
Comment (topolyai5) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: use underscore

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java [hunk 5]
```diff
@@ -28,66 +27,66 @@
...
             nickname = "deleteEnvironment")
-    SimpleEnvironmentResponse delete(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName);
+    SimpleEnvironmentV4Response delete(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName);
 
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.LIST, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "listEnvironment")
-    Set<SimpleEnvironmentResponse> list(@PathParam("workspaceId") Long workspaceId);
+    SimpleEnvironmentResponses list(@PathParam("workspaceId") Long workspaceId);
```
Comment (horadla23) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: add V4 to the name of this class as well

**Pull request merged**